virtualenv_tools.py

```diff
@@ -91,17 +91,35 @@ def update_script(
             return
         f.seek(0)
         lines = list(f)
-    args = lines[0][2:].strip().split()
-    if not args:
-        return
 
-    if path_is_within(args[0], old_path):
-        new_bin = os.path.join(new_path, os.path.relpath(args[0], old_path))
+    # is this a python script being run under a bourne exec call
+    if lines[0].strip() == b'#!/bin/sh' and lines[1].startswith(b"'''exec' "):
+        args = lines[1].strip().split()
```

Comment:
this IndexErrors on a file with only one line

Reply:
Yep, I'll add a len(lines) check first.